src/Chapter3.hs

```diff
@@ -479,6 +500,39 @@ After defining the city, implement the following functions:
    complicated task, walls can be built only if the city has a castle
    and at least 10 living people inside
 -}
+data City = City
+  { cityCastle :: Castle
+  , cityMain   :: MainBuilding
+  , cityHouses :: [House]
+  }
+
+data Castle = None | OnlyCastle | CastleWithWalls
```

Comment:
The task tells that the castle should have a name. In this case, it can be a `String`. Sorry, if it wasn't clear before. We improved this task wording in the latest version to clarify this moment.